### DISCUSSION TITLE
How can I prevent duplicate file uploads?

### DISCUSSION THREAD

**Discussion by Xyphyn** [pocketbase/pocketbase#1929] [Q&A]
I want to prevent people from spamming posts with the same image. My first plan was to check "unique" in the image parameter, but it isn't available for files. What should I do?

Also, it should be much easier to delete an account.

**Comment by ganigeorgiev** [comment]
There is no built-in files unique check at the moment. If you want to prevent duplicated files from being uploaded you have to generate a hash based on their content (eg. md5) (and maybe store it somewhere alongside the file to prevent the performance overhead of generating it every time). For that you'll have to use the [`OnRecordBefore*` hooks](https://pocketbase.io/docs/event-hooks/#record-crud-api-hooks).

> Also, it should be much easier to delete an account.

Could you elaborate a little more? Deleting an auth record via the JS SDK is similar to any other collection record:

```js
await pb.collection("users").delete("RECORD_ID")
```

**Reply by Xyphyn** [reply]
The last remark was when somebody spammed my database with a bunch of duplicate records. Since then I learned about cascade delete. I've also implemented rate limits.
